packages/terra-tabs/src/Tabs.jsx

```diff
@@ -151,98 +85,43 @@ class Tabs extends React.Component {
-      { 'tab-fill': tabFill },
-      variant,
-      theme.className,
-    ),
-    customProps.className);
-
-    let content = null;
-    let isIconOnly = false;
-    const clonedPanes = [];
-    React.Children.forEach(children, (child, index) => {
-      let isActive = false;
-      if (child.key === this.state.activeKey || child.key === activeKey) {
-        isActive = true;
-        content = child.props.children;
-      }
+    const commonTabItems = [];
 
-      if (child.props.isIconOnly) {
-        isIconOnly = true;
+    React.Children.forEach(children, child => {
+      let content;
+      let workspaceContent;
+      if (child.key === this.state.activeKey) {
+        content = React.Children.map(child.props.children, contentItem => (
+          React.cloneElement(contentItem)
+        ));
+        workspaceContent = <CommonTabContent>{content}</CommonTabContent>;
       }
-      clonedPanes.push(React.cloneElement(child, {
-        className: cx({ 'is-active': isActive }, child.props.className),
-        onClick: this.wrapPaneOnClick(child),
-        isActive,
-        index,
-      }));
-    });
-
-    content = React.Children.map(content, contentItem => (
-      React.cloneElement(contentItem, { isLabelHidden: isIconOnly || this.state.isLabelTruncated })
-    ));
-
-    const tabContent = (() => {
-      const collapsibleTabs = (
-        <CollapsibleTabs
-          activeKey={activeKey || this.state.activeKey}
-          activeIndex={this.getActiveTabIndex()}
-          onChange={this.handleOnChange}
-          onTruncationChange={this.handleTruncationChange}
-          variant={variant}
-        >
-          {clonedPanes}
-        </CollapsibleTabs>
+      commonTabItems.push(
+        <CommonTabItem
+          itemKey={child.key}
+          label={child.props.label}
+          icon={child.props.icon}
+          isIconOnly={child.props.isIconOnly}
+          render={() => workspaceContent}
+          isDisabled={child.props.isDisabled}
+        />,
       );
-
-      if (responsiveTo === 'parent' || responsiveTo === 'window') {
-        const collapsedTabs = (
-          <CollapsedTabs
-            activeKey={activeKey || this.state.activeKey}
-            activeIndex={this.getActiveTabIndex()}
-            onTruncationChange={this.handleTruncationChange}
-          >
-            {clonedPanes}
-          </CollapsedTabs>
-        );
-
-        return (
-          <ResponsiveElement
-            onChange={this.setTabs}
-            responsiveTo={responsiveTo}
-          >
-            { this.state.showCollapsedTabs ? collapsedTabs : collapsibleTabs}
-          </ResponsiveElement>
-        );
-      }
-
-      return collapsibleTabs;
-    })();
+    });
 
     return (
-      <ContentContainer
+      <CommonTabs
+        id={customProps.id}
+        activeItemKey={this.state.activeKey}
+        onRequestActivate={key => this.setState({ activeKey: key })}
+        onChange={onChange}
+        variant="framework"
```

Comment:
use a constant.

Reply:
The name and value seem self-explanatory here so it feels redundant to have a constant.